applications/HDF5Application/tests/test_hdf5_core_mpi.py

```diff
@@ -27,26 +49,22 @@ def test_FilenameGetterWithDirectoryInitialization_DirectoryExists(self, mock_cl
         mock_instance = mock_class.return_value
         mock_instance.GetFileName.return_value = '/foo/kratos.h5'
         settings = self._FilenameGetterSettings(file_name='/foo/kratos.h5')
-        patcher = patch('KratosMultiphysics.FilesystemExtensions.MPISafeCreateDirectories', autospec=True)
-        makedirs = patcher.start()
-        data_comm = KratosMultiphysics.Testing.GetDefaultDataCommunicator()
-        obj = file_io._FilenameGetterWithDirectoryInitialization(settings, data_comm)
-        obj.Get(_SurrogateModelPart())
-        makedirs.assert_called_once_with('/foo')
-        patcher.stop()
+        with patch('KratosMultiphysics.FilesystemExtensions.MPISafeCreateDirectories', autospec=True) as makedirs:
+            data_comm = KratosMultiphysics.Testing.GetDefaultDataCommunicator()
+            obj = file_io._FilenameGetterWithDirectoryInitialization(settings, data_comm)
+            obj.Get(_SurrogateModelPart()[1])
+            makedirs.assert_called_once_with('/foo')
```

Comment:
I didn't change this test apart from scoping the patcher such that it doesn't bleed outside the function if there's an assertion failure.

Do you want me to change the original test to use `MPISafeCreateDirectories` directly instead of mocking it?